tests/unit/s2n_renegotiate_io_test.c

```diff
@@ -0,0 +1,483 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#include "tls/s2n_renegotiate.h"
+
+#include "tls/s2n_tls.h"
+#include "utils/s2n_safety.h"
+#include "utils/s2n_socket.h"
+
+enum S2N_TEST_APP_DATA_CASES {
+    S2N_TEST_APP_DATA_BEFORE_RENEG          = 1,
+    S2N_TEST_APP_DATA_BEFORE_SERVER_HELLO   = 2,
+    S2N_TEST_APP_DATA_AFTER_SERVER_HELLO    = 4,
+    S2N_TEST_MAX_TEST_CASES                 = 8,
+};
+#define S2N_TEST_APP_DATA_LEN 10
+
+int main(int argc, char *argv[])
+{
+    BEGIN_TEST();
+
+    DEFER_CLEANUP(struct s2n_cert_chain_and_key *chain_and_key = NULL, s2n_cert_chain_and_key_ptr_free);
+    EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+            S2N_DEFAULT_TEST_CERT_CHAIN, S2N_DEFAULT_TEST_PRIVATE_KEY));
+
+    DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+    EXPECT_NOT_NULL(config);
+    EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(config));
+    EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key));
+    EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "default"));
+
+    uint8_t app_data[] = "smaller hello world";
+    //uint8_t large_app_data[S2N_TLS_MAXIMUM_FRAGMENT_LENGTH] = "hello world and a lot of zeroes";
```

Comment:
?

Reply:
Copy / pasting from another test -_- I've now removed the line.